services/src/main/java/org/jboss/windup/web/services/rest/WindupEndpoint.java

```diff
@@ -28,12 +31,20 @@
     WindupExecution getExecution(@PathParam("executionID") Long executionID);
 
     /**
-     * Initiates a Windup execution for a particular AnalysisContext.
+     * Initiates a Windup execution for the given existing AnalysisContext.
      */
     @POST
     @Path("execute-with-context")
     WindupExecution executeWithContext(Long contextID);
 
+    /**
+     * Initiates a Windup execution for the given AnalysisContext.
+     * The context is persisted as a new instance, and attached to the new execution.
+     */
+    @POST
+    @Path("execute-with-context/{projectId}")
+    WindupExecution executeProjectWithContext(@Valid AnalysisContext analysisContext, @PathParam("projectId") @NotNull Long projectId);
+
```

Comment:
Could you please explain your first point more? I don't think we should override one endpoint to do different thing based on parameters.
Regarding the second, you're right, I'm renaming it to execute-project-with-context.